entrypoint.sh

```diff
@@ -12,10 +12,19 @@ if echo ${@} | grep -q "cloud-provider=azure"; then
   fi
 fi
 
+# find docker root
+DOCKER_ROOT="/var/lib/docker"
+if [ -f "/etc/docker/daemon.json" ]; then
+  CONFIGURED_ROOT=$(grep -oP '(?<="data-root":\ ")(.*)(?=")' /etc/docker/daemon.json)
+  if [ -n "${CONFIGURED_ROOT}" ]; then
+    DOCKER_ROOT=CONFIGURED_ROOT
+  fi
+fi
+
 if [ "$1" = "kubelet" ]; then
     for i in $(DOCKER_API_VERSION=1.24 /opt/rke-tools/bin/docker info 2>&1  | grep -i 'docker root dir' | cut -f2 -d:) /var/lib/docker /run /var/run; do
         for m in $(tac /proc/mounts | awk '{print $2}' | grep ^${i}/); do
-            if [ "$m" != "/var/run/nscd" ] && [ "$m" != "/run/nscd" ]; then
+            if [ "$m" != "/var/run/nscd" ] && [ "$m" != "/run/nscd" ] && [ "$m" != "${DOCKER_ROOT}/containers" ]; then
```

Comment:
This solves `/containers` but the user can have any subdir of docker root on a separate mount, did we check the command to see if the sub dir is a mount? (e.g. `mountpoint`)